weasyl/test/test_login.py

```diff
@@ -47,14 +47,14 @@ def __init__(self, **kw):
 def testSignin_VerifyLoginRecordIsUpdated():
     user_name = TestFunctions().generateTestAccountName()
     user_id = db_utils.create_user(username=user_name)
-    d.execute("UPDATE login SET last_login = '-1' WHERE userid = %i", [user_id])
+    d.engine.execute("UPDATE login SET last_login = '-1' WHERE userid = %(id)s", id=user_id)
     # login.signin(user_id) -will- raise an AttributeError when d.web.ctx.weasyl_session
     #   tries to execute itself; so catch/handle; it's a test environment issue
     with pytest.raises(AttributeError) as err:
         login.signin(user_id)
     print str(err)
-    query = d.execute("SELECT last_login FROM login WHERE userid = %i", [user_id])
-    assert query[0][0] > -1
+    query = d.engine.execute("SELECT last_login FROM login WHERE userid = %(id)s", id=user_id)
+    assert query.first()[0] > -1
```

Comment:
`query.first()[0]` → `query.scalar()`, or better, `last_login = d.engine.scalar(…)` so the variable name is descriptive.